docker.go

```diff
@@ -1414,3 +1414,67 @@ func (p *DockerProvider) getDefaultNetwork(ctx context.Context, cli client.APICl
 
 	return reaperNetwork, nil
 }
+
+// ListImages list images from the provider. If an image has multiple Tags, each tag is reported
+// individually with the same ID and same labels
+func (p *DockerProvider)ListImages(ctx context.Context) ([]ImageInfo, error) {
+	images := []ImageInfo{}
+
+	imageList, err := p.client.ImageList(ctx, types.ImageListOptions{})
+	if err != nil {
+		return images, fmt.Errorf("listing images %w", err)
+	}
+
+	for _, img := range imageList {
+		for _, tag := range img.RepoTags {
+			images = append(images, ImageInfo{ID: img.ID, Name: tag})
+		}
+	}
+
+	return images, nil
+}
+
+
+// SaveImages exports a list of images as an uncompressed tar
+func (p *DockerProvider)SaveImages(ctx context.Context, output string, images...string) error  {
+	outputFile, err := os.Create(output)
+	if err != nil {
+		return fmt.Errorf("opening output file %w", err)
+	}
+	defer func(){
+		_ = outputFile.Close()
+	}()
+
+	imageReader, err := p.client.ImageSave(ctx, images)
+	if err != nil {
+		return fmt.Errorf("saving images %w", err)
+	}
+	defer func(){
+		_ = imageReader.Close()
+	}()
+
+	_, err = io.Copy(outputFile, imageReader)
+	if err != nil {
+		return fmt.Errorf("writing images to output %w", err)
+	}
+
+	return nil
+}
+
+// PullImage pulls image from registry
+func (p *DockerProvider)PullImage(ctx context.Context, image string) error {
```

Comment:
Could you reuse the code block in https://github.com/testcontainers/testcontainers-go/pull/1622/files#diff-b68159b6168f099a639e8c6f96d75f23333ab0f0505ee22a98268748db17ae4cR1155-R1184 into this PullImage method? It basically does the same but with Retries.

We could remove the `attempToPullImage` method and replace all its usages for this new method (including the old comments).

Thoughts?

Reply:
It makes sense to have only one method. I will do that.

Reply:
The only inconvenience I see in renaming `attempToPullImage` is exposing the `types.ImagePullOptions` argument which comes from the docker client package. Maybe `PullImage` should wrap it and pass a default ImagePullOptions?

Reply:
Makes sense to me, thanks for considering the downstream usage of the API 🙏